### ISSUE TITLE
bug: import statements from dist-custom-elements getting passed to commonjs and errors in SSR Applications like NextJS.

### ISSUE THREAD

**Issue by roman-telia** [ionic-team/stencil#3176]
### Prerequisites

- [X] I have read the [Contributing Guidelines](https://github.com/ionic-team/stencil/blob/main/.github/CONTRIBUTING.md).
- [X] I agree to follow the [Code of Conduct](https://github.com/ionic-team/stencil/blob/main/CODE_OF_CONDUCT.md).
- [X] I have searched for [existing issues](https://github.com/ionic-team/stencil/issues) that already report this problem, without success.

### Stencil Version

2.11.0

### Current Behavior

after we updated from dist-custom-elements-bundle to dist-custom-elements we are running into issues with NextJS when using our commonjs react wrappers in NextJS app, when running next js or building it throws error that can't use import statement outside of a module.

### Expected Behavior

It should work the same way as before with dist-custom-elements-bundle, that there wasn't imported es6 module files in commonjs react output.

### Steps to Reproduce

create build output, install new package version in Next js project, try to run the next app after installing latest package and you will get this error.

### Code Reproduction URL

https://github.com/roman-telia/Next-vocads-playground

**Comment by rwaskiewicz** [comment]
@roman-telia Thanks for the issue! I don't seem to have access to that reproduction url - would you be able to give me access (or create a separate public repo)?

**Comment by roman-telia** [comment]
Hi rwaskiewicz, i have created github repository as public one, as you requested.

**Comment by nilssonja** [comment]
This is something that my team has been experiencing for some time when using the `dist-custom-elements` output target. Our setup is as follows:

We have our "main" library that uses both the `dist` output target and the `dist-custom-elements` output target. Since we use both output targets, our package.json for the library set's the `module`, `es2015`, `es2017`, etc to the files generated by the `dist` output target. This means that consumers of our main library by default use the lazy-loaded `dist` components and must manually import the `dist-custom-elements` output from `dist/components/<component_name>.js` if they instead want to manually define and consume our components.

Next we are using the React output target to generate a React library that consumes the `dist-custom-elements` output and NOT the `dist` output. We do this for a number of reasons, namely we have applications with strict bundle size concerns and wanted a way to ensure ALL code we ship to them is tree-shakable all the way down to the individual web components.

Onto the crux of the issue: Because the `dist-custom-elements` output target does not transform into cjs, we found React consumers complaining of the `SyntaxError: Cannot use import statement outside a module` error within their jest tests. We've worked around this in the meantime by suggesting using `transformIgnorePatterns` in their jest/babel config. However, we are now seeing new Next.JS applications spin up who are running into the issue described above.

Unlike with the `dist` output target, is it up to us to come up with a custom bundling solution for `dist-custom-elements` separate from Stencil's build process for teams who need to consume them in the cjs format?
